### PR TITLE
feat(data): update PR template with E2E test check for code examples; update docs link example to point to an Amplify-owned resource

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Description of changes:
While adding TypeScript examples to our DataStore docs, I noticed several existing code examples that were broken. See:
https://github.com/aws-amplify/docs/issues/5156
https://github.com/aws-amplify/docs/issues/5157
https://github.com/aws-amplify/docs/issues/5168
https://github.com/aws-amplify/docs/issues/5231
https://github.com/aws-amplify/amplify-js/issues/11006
https://github.com/aws-amplify/amplify-js/issues/9618

Related docs fixes:
https://github.com/aws-amplify/docs/pull/5155
https://github.com/aws-amplify/docs/pull/5140

Updating our PR template with a check for new / updated E2E tests for all new / updated code examples in the docs.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [x] amplify-cli
- [x] amplify-ui
- [x] amplify-studio
- [x] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [x] iOS
- [x] Android
- [x] Flutter
- [x] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br />
      _ref: MDX: `[link](https://link.com)`
            HTML: `<a href="https://link.com">link</a>`_
            
### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -30,11 +32,12 @@ Which platform(s) are affected by this PR (if applicable)?
 
 - [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?
 
-- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br />
-      _ref: MDX: `[link](https://link.com)`
-            HTML: `<a href="https://link.com">link</a>`_
-            
+- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> _ref: MDX: `[link](https://link.com)` HTML: `<a href="https://link.com">link</a>`_
```
Comment (david-mcafee) [inline .github/PULL_REQUEST_TEMPLATE.md]: `link.com` is an actual website, but not owned by Amplify.

**Review by jimblanc — DISMISSED** [review]

**Review by stocaaro — APPROVED** [review]

**Review by rachelnabors — CHANGES_REQUESTED** [review]
I have questions... How are non-Amplify team members expected to participate in this? Is there a way to permanently associate a test with a code snippet? How does creating E2E tests for code snippets impact our writing guidelines?

**Comment by david-mcafee** [comment]
> I have questions... How are non-Amplify team members expected to participate in this? Is there a way to permanently associate a test with a code snippet? How does creating E2E tests for code snippets impact our writing guidelines?

@rachelnabors - when you say `non-Amplify team members`, do you mean external contributors? If so, we would not expect them to write E2E tests. However, we _would_ expect an Amplify team member to add / update the E2E tests as necessary.

`Is there a way to permanently associate a test with a code snippet?` - we don't currently have a mechanism for this that I'm aware of, at least with regards to the JS library, though it's something we can certainly explore.

`How does creating E2E tests for code snippets impact our writing guidelines?` - if you are referring to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md), I can update that, as well.

Let me know if you have further questions!

**Comment by rachelnabors** [comment]
I see. I recommend that, rather than adding another tick box that doesn't mean anything to external contributors, we add mechanisms to ensure each line of code added has an associated test, like we use to ensure PRs that remove files have redirects. That would be a good part of a project we're working on. I'll put you in touch with the right folks on the effort.

**Comment by hbuchel** [comment]
Pulled out the fix for updating the `link.com` example link into [this separate PR](https://github.com/aws-amplify/docs/pull/6730). 

Closing this PR due to its age and the docs site has since been refactored. Please feel free to update with latest and re-open if you feel it is still relevant, thanks.

**Pull request closed without merging**